rules/fusesoc.bzl

```diff
@@ -26,12 +26,11 @@ def _corefiles2rootarg(core):
 
 def _fusesoc_build_impl(ctx):
     dirname = "build.{}".format(ctx.label.name)
-    out_dir = ctx.actions.declare_directory(dirname)
     flags = [ctx.expand_location(f, ctx.attr.srcs) for f in ctx.attr.flags]
-    outputs = [out_dir]
+    outputs = []
```

Comment:
If we're getting rid of the directory here, we should probably add at least the routed design checkpoint (DCP) to all of the bitstream build targets' outputs. Otherwise, we won't be able to readily analyze the quality of results and adjust constraints.

Reply:
I've found a couple more paths from the bitstream build that are used, e.g. `ci/fpga-template.yml` tries to read some `runme.log` files which are no longer present.

I'll think about what to do with these

Reply:
I've replaced the implementation with a symlink one:

* The `build` directory is left alone and included in the outputs.
* Symlinks are created in an `out` directory pointing into `build`.
* The symlinks are used as the `output_groups`.

Reply:
Update: had to revert the symlink idea, will think of some other way to resolve this.

Reply:
For what it's worth, it seems fine to get rid of the build directory as an output for DefaultInfo, but at each fusesoc_build() target, we'll want to be less lazy and add the other specific files we need to keep, hehe. I think the directory-as-output was intended more for generated files with unpredictable names, so we're probably abusing it.

We could also modify our Tcl scripts (and possibly fusesoc core file) to rename the output files to include the label in the Bazel BUILD file. That's just an aesthetic change, though.

Reply:
I've added the `runme.log` and final DCP file to the outputs of our Vivado rules. Hopefully this is enough. I had problems with name collisions adding the `impl_1/runme.log` file so have removed this one for now. It will take some slightly larger changes to fix that.